src/Analyzer.TemplateProcessor/ArmTemplateProcessor.cs

```diff
@@ -124,39 +125,29 @@ internal Template ParseAndValidateTemplate(InsensitiveDictionary<JToken> paramet
 
             TemplateEngine.ValidateProcessedTemplate(template, apiVersion, TemplateDeploymentScope.NotSpecified);
 
-            template = ProcessTemplateResourceAndOutputPropertiesLanguageExpressions(template);
+            template = ProcessResourcesAndOutputs(template);
 
             return template;
         }
 
         /// <summary>
-        /// Processes language expressions in the properties property of the resources and value property of outputs.
+        /// Processes each resource for language expressions and parent resources as well
+        /// as processes language expressions for outputs.
         /// </summary>
-        /// <param name="template">The template.</param>
-        /// <returns>The parsed template as a Template object.</returns>
-        internal Template ProcessTemplateResourceAndOutputPropertiesLanguageExpressions(Template template)
+        /// <param name="template">Template being processed.</param>
+        /// <returns>Template after processing resources and outputs.</returns>
+        internal Template ProcessResourcesAndOutputs(Template template)
         {
             var evaluationHelper = GetTemplateFunctionEvaluationHelper(template);
+            var flattenedResources = FlattenResources(template.Resources);
 
             for (int i = 0; i < template.Resources.Length; i++)
             {
                 var resource = template.Resources[i];
 
-                try
-                {
-                    if (resource.Properties != null)
-                    {
-                        evaluationHelper.OnGetCopyContext = () => resource.CopyContext;
-                        resource.Properties.Value = ExpressionsEngine.EvaluateLanguageExpressionsRecursive(
-                            root: resource.Properties.Value,
-                            evaluationContext: evaluationHelper.EvaluationContext);
-                    }
-                }
-                catch
-                {
-                    // Do not throw if there was an issue with evaluating language expressions
-                    continue;
-                }
+                ProcessTemplateResourceLanguageExpressions(resource, evaluationHelper);
+
+                CopyResourceDependants(resource, flattenedResources);
```

Comment:
If the call to ProcessTemplateResourceLanguageExpressions fails, won't CopyResourceDependant have problems and potentially throw an unexpected exception?

I think it could definitely happen if the return value of  ProcessTemplateResourceLanguageExpressions is meant to be used

Reply:
Since there we're catching all exceptions in ProcessTemplateResourceLanguageExpressions I don't think so... ProcessTemplateResourceLanguageExpressions doesn't need to return anything since  all the changes it makes are by reference anyway. I can't think of a case where ProcessTemplateResourceLanguageExpressions eats an exception there would be issues in CopyResourceDependants.